### PR TITLE
[Navigation] Consolidate se23 logic and styles

### LINKED ISSUE #9952: [Navigation] Consolidate se23 logic and styles

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Resolves #9952.

### WHAT is this pull request doing?

Consolidates se23 beta styles and logic for `Navigation`.

### How to 🎩
[Storybook](https://5d559397bae39100201eedc1-lecmsuzwza.chromatic.com/?path=/story/all-components-navigation--default)
[Prod Storybook](https://storybook.polaris.shopify.com/?path=/story/all-components-navigation--default&globals=polarisSummerEditions2023:true)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -379,7 +315,7 @@ $nav-max-width: 360px;
     // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
     @include no-focus-ring;
     color: var(--p-color-text-primary);
-    background-color: var(--p-color-bg-app-active);
+    background-color: var(--p-color-bg-app-selected);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:318]: Changed due to specificity issues.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 2]
```diff
@@ -342,30 +288,20 @@ $nav-max-width: 360px;
...
+  .Text {
+    font-weight: var(--p-font-weight-semibold);
   }
 
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   @include focus-ring;
 
   &:hover,
   &:focus-visible {
-    background-color: var(--p-color-bg-app-hover);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:368]: Removed due to specificity issues.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 3]
```diff
@@ -427,31 +360,18 @@ $nav-max-width: 360px;
...
   }
 }
 
 .ListItem-hasAction .Badge {
-  margin-right: var(--p-space-5);
-
-  @media #{$p-breakpoints-md-up} {
-    margin-right: var(--p-space-3);
-  }
+  margin-right: var(--p-space-1);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:374]: Removed breakpoint margin styling due to specificity issues.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 4]
```diff
@@ -520,42 +424,23 @@ $nav-max-width: 360px;
...
   display: flex;
   flex-wrap: nowrap;
-
-  #{$se23} & {
-    width: 100%;
-  }
-
-  .SecondaryActions {
-    margin-right: 0;
-  }
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:558]: Removed this margin due to specificity issues.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 5]
```diff
@@ -690,7 +544,6 @@ $nav-max-width: 360px;
   &:active {
     // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
     @include no-focus-ring;
-    background: var(--p-color-bg-app-active);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:693]: Removed due to specificity issues.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 6]
```diff
@@ -720,68 +573,53 @@ $nav-max-width: 360px;
...
+        font-weight: var(--p-font-weight-medium);
+        // stylelint-disable-next-line -- custom property
+        letter-spacing: var(--pc-navigation-letter-spacing-medium);
       }
     }
 
     &:focus-visible {
       // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
       @include focus-ring($style: 'focused');
-      color: var(--p-color-text);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:773]: Removed due to specificity issues.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 7]
```diff
@@ -791,51 +629,40 @@ $nav-max-width: 360px;
     @media #{$p-breakpoints-md-up} {
       font-size: var(--p-font-size-100);
       line-height: 1;
-      padding-left: var(--p-space-10);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:794]: Removed due to specificity issues.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 8]
```diff
@@ -1128,6 +922,10 @@ $nav-max-width: 360px;
     // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
     @include focus-ring($style: 'focused');
   }
+
+  &:active {
+    background-color: var(--p-color-bg-app-active);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/Navigation.scss:927]: Added due to specificity issues.

File: polaris-react/src/components/Navigation/_variables.scss [hunk 9]
```diff
@@ -46,31 +46,26 @@ $nav-animation-variables: (
...
+    z-index: var(--p-z-index-1);
+    outline: var(--p-border-width-2) solid
+      var(--p-color-border-interactive-focus);
+  }
+
   &:active,
   &:active:hover {
     color: var(--p-color-text);
-    background-color: var(--p-color-bg-app-active);
+    background-color: var(--p-color-bg-app-selected);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/_variables.scss:60]: Changed due to specificity issues.

File: polaris-react/src/components/Navigation/_variables.scss [hunk 10]
```diff
@@ -46,31 +46,26 @@ $nav-animation-variables: (
...
-      outline: var(--p-border-width-2) solid
-        var(--p-color-border-interactive-focus);
-    }
-  }
-
   @media #{$p-breakpoints-md-up} {
     font-size: var(--p-font-size-200);
     font-weight: var(--p-font-weight-medium);
     line-height: var(--p-font-line-height-5);
-    padding-left: var(--p-space-3);
```
Comment (laurkim) [inline polaris-react/src/components/Navigation/_variables.scss:73]: Removed due to specificity issues.

**Review by sophschneider — APPROVED** [review]
🤩

**Comment by sophschneider** [comment]
btw I tophatted xs screen, keyboard navigation, and responsive screen and everything had parity with production! I did notice some bugs in production on xs screens, I've logged them here https://github.com/Shopify/polaris/issues/10218 since it isn't related to this consolidation

**Pull request merged**